src/lando/ui/jinja2/jobs/job.html

```diff
@@ -0,0 +1,39 @@
+{% extends "partials/layout.html" %}
+{% block page_title %}Landing Job {{job.id}} - Lando - Mozilla{% endblock %}
```

Comment:
nit: may be time to move `Lando - Mozilla` to outside this block so that it's not repeated in multiple templates.